lib/spack/spack/test/concretization/core.py

```diff
@@ -1460,6 +1460,19 @@ def test_no_reuse_when_variant_condition_does_not_hold(self, mutable_database, m
         new2 = spack.concretize.concretize_one("conditional-variant-pkg +two_whens")
         assert new2.satisfies("@2 +two_whens +version_based")
 
+    def test_reuse_by_namespace(self, mutable_database, mock_packages):
+        spack.config.set("concretizer:reuse", True)
+        spack.config.set("packages:libelf", {"version": ["0.8.10"]})
+
+        reuse = spack.concretize.concretize_one("libelf")
+        assert reuse.installed
+        assert reuse.satisfies("@0.8.13")
```

Comment:
Even with reuse on, shouldn't specifying then exact version force using that version? I would not expect the behavior checked here.

Reply:
Strike that. This needs a better comment about why this magic version is expected.

I think this probably shouldn't point at the `builtin` namespace as we are actively trying to remove that from Spack core. Relying on that in unit tests is maybe not the best option as it will be free to change independent of Spack over time.

@tgamblin was talking today about using the other mock/test repos we have in Spack for this kind of thing.

Reply:
This does not point at the builtin repo, it points at `builtin.mock`, which we use extensively throughout the tests. That's what the `mock_packages` fixture in pytest does.

Reply:
I've added a comment clarifying that we check for `libelf@0.8.13` because that's the one installed in the database from the `mutable_database` fixture.